include/boost/math/statistics/linear_regression.hpp

```diff
@@ -46,35 +50,39 @@ auto simple_ordinary_least_squares(RandomAccessContainer const & x,
     return std::make_pair(c0, c1);
 }
 
-template<class RandomAccessContainer>
-auto simple_ordinary_least_squares_with_R_squared(RandomAccessContainer const & x,
-                                   RandomAccessContainer const & y)
+template<typename RandomAccessContainer, typename Real = typename RandomAccessContainer::value_type, 
+         typename Size = typename RandomAccessContainer::size_type>
+auto simple_ordinary_least_squares_with_R_squared(RandomAccessContainer const & x, RandomAccessContainer const & y)
+    -> std::tuple<Real, Real, Real>
 {
-    using Real = typename RandomAccessContainer::value_type;
-    if (x.size() <= 1)
+    if (std::size(x) <= 1)
```

Comment:
I don't have a problem with `std::size`, and in fact it solves a real problem. But it also has historically caused problems which is why I avoid it.

Did you have a type that requires `std::size`?

Reply:
I have no compelling reason other than Scott Myers et. al. recommend it, and I was already changing things.

Reply:
Ok, maybe the ecosystem has improved a bit since I first tried it. I'm fine with this change.